core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -74,7 +74,7 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
   val pollTask: Cancellable =
     context.system.scheduler.schedule(pollInterval(), pollInterval(), self, Poll)(context.dispatcher)
 
-  var requests = Map.empty[TopicPartition, ActorRef]
+  var requests = Map.empty[ActorRef, RequestMessages]
```

Comment:
@patriknw with current implementation it is possible to reply to two different actors with same messages. It should not happens from my point of view